lib/puppet/provider/service/init.rb

```diff
@@ -61,6 +61,11 @@ def self.excludes
     excludes += %w{cryptdisks-udev}
     excludes += %w{statd-mounting}
     excludes += %w{gssd-mounting}
+    # Prevent puppet failing on unsafe scripts from Yocto Linux
+    excludes += %w{banner.sh bootmisc.sh checkroot.sh devpts.sh dmesg.sh
+                 hostname.sh mountall.sh mountnfs.sh populate-volatile.sh
+                 rmnologin.sh save-rtc.sh sendsigs sysfs.sh umountfs
+                 umountnfs.sh}
```

Comment:
From what I've seen a fair amount of these do exist on Debian based systems, but these scripts are relegated to boot and shutdown time.

In particular, some of the shutdown scripts behave okay on Deb when you try and status them, but they are definitely shutdown only bits.

* sendsigs
 * Kill all remaining processes.
* umountfs
  * Turn off swap and unmount all local file systems.
* umountnfs.sh
  * Unmount all network filesystems except the root fs.

Examples of the boot ones
* bootmisc.sh
  * Miscellaneous things to be done during bootup.
* hostname.sh
  * Set hostname based on /etc/hostname
* mountall.sh
  * Mount all filesystems.
* mountnfs
  * Wait for network file systems to be mounted
* rmnologin
  * Remove /etc/nologin at boot